int32 chamber
int32[] wall
bool state
bool send
